ssg/variables.py

```diff
@@ -0,0 +1,415 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+import glob
+import yaml
+
+from collections import defaultdict
+from .constants import BENCHMARKS
+from .controls import ControlsManager
+from .products import (
+    get_profile_files_from_root,
+    load_product_yaml,
+    product_yaml_path,
+)
+from .yaml import open_and_macro_expand
+
+# Cache variable files to avoid multiple reads
+_var_files_cache = {}
+
+def get_variable_files_in_folder(content_dir: str, subfolder: str) -> list[str]:
+    """
+    Retrieve a list of variable files within a specified folder in the project.
+
+    Args:
+        content_dir (str): The root directory of the content.
+        subfolder (str): The folder within the project to search for variable files.
+
+    Returns:
+        list: A list of paths to variable files found within the specified folder.
+    """
+    rules_base_dir = os.path.join(content_dir, subfolder)
+    search_dir = os.path.normpath(rules_base_dir)
+    pattern = os.path.join(search_dir, '**', '*.var')
+    return glob.glob(pattern, recursive=True)
+
+
+def get_variable_files(content_dir: str) -> list[str]:
+    """
+    Retrieves all variable files from the specified content root directory.
+
+    This function iterates through a predefined list of benchmark directories and collects all
+    variable files within those directories.
+
+    Args:
+        content_dir (str): The root directory containing benchmark directories.
+
+    Returns:
+        list: A list of paths to all variable files found within the benchmark directories.
+    """
+    if content_dir in _var_files_cache:
+        return _var_files_cache[content_dir]
+
+    variable_files = []
+    for benchmark_directory in BENCHMARKS:
+        variable_files.extend(get_variable_files_in_folder(content_dir, benchmark_directory))
+    _var_files_cache[content_dir] = variable_files
+    return variable_files
+
+
+def get_all_variables_options(content_dir: str) -> dict[str, dict]:
```

Comment:
Do you think you could somehow reuse the function get_variable_options here? I think the code duplicity is not necessary. Both functions have very similar for loops.

Reply:
Let me see what I can do here.

Reply:
Could you take a look again, please? More details in the commit message.